bundler/lib/bundler/lockfile_parser.rb

```diff
@@ -136,7 +139,13 @@ def initialize(lockfile)
         end
         @pos.advance!(line)
       end
-      @specs = @specs.values.sort_by!(&:full_name)
+      @most_specific_locked_platform = @platforms.min_by do |bundle_platform|
+        platform_specificity_match(bundle_platform, local_platform)
+      end
+      @specs = @specs.values.map do |spec|
```

Comment:
nit: since you're doing `specs.values`, which returns a new array each time, you can `map!` and `sort_by!` to save a pair of allocations

actually, `map` isn't mapping anything, so it can be a `sort_by!` followed by an `each` to set `most_specific_locked_platform`

Reply:
Thank you, that makes sense!

Actually at some point I had changed this to leave instantiation of all `LazySpecification`'s directly here, which I think would simplify the code a bit a maybe save a allocations too. I can push that for you to have a look.

Reply:
I added just with suggestion, because my idea was not as easy as I thought.